### PR TITLE
support vip dual stack

### PR DESCRIPTION
**Pull request by zcq98**
# Pull Request

- [x] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

## What type of this PR

Examples of user facing changes:

- Features

## WHAT

copilot:summary

copilot:poem

## HOW

copilot:walkthrough

### CODE REVIEW

File: pkg/controller/pod.go [hunk 1]
```diff
@@ -2030,10 +2030,11 @@ func (c *Controller) getVirtualIPs(pod *v1.Pod, podNets []*kubeovnNet) map[strin
 				continue
 			}
 			key := fmt.Sprintf("%s.%s", subnet.Name, subnet.Spec.Provider)
-			if vips, exists := vipsMap[key]; exists {
-				vipsMap[key] = strings.Join([]string{vips, vip.Status.V4ip}, ",")
+			ipStr := util.GetStringIP(vip.Status.V4ip, vip.Status.V6ip)
+			if vips, ok := vipsMap[key]; ok {
+				vipsMap[key] = strings.Join([]string{vips, ipStr}, ",")
```
Comment (oilbeater) [inline pkg/controller/pod.go:2035]: 是不是需要考虑一下 vips 和 ipStr 可能有重叠

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: pkg/controller/pod.go (hunk 1 above)
Comment (zcq98) [reply pkg/controller/pod.go]: 已修复

File: pkg/controller/pod.go [hunk 2]
```diff
@@ -2013,42 +2013,65 @@ func getPodType(pod *v1.Pod) string {
...
+	vipsMap := make(map[string]string)
+	for key, vipsList := range vipsListMap {
+		vipsMap[key] = strings.Join(vipsList, ",")
 	}
 	return vipsMap
 }
+
+func addVIPToList(vipsList []string, vip string) []string {
+	if util.IsValidIP(vip) {
+		for _, v := range vipsList {
```
Comment (oilbeater) [inline pkg/controller/pod.go:2069]: 可以使用 `util.ContainsString` 简化逻辑

**Review by oilbeater — CHANGES_REQUESTED** [review]

File: pkg/controller/pod.go (hunk 2 above)
Comment (zcq98) [reply pkg/controller/pod.go]: 已修改，针对AAPsAnnotation注解，仅需对vipName去重， ipam会保证vip的唯一性；针对PortVipAnnotation则需要util.ContainsString进行VIP去重

**Review by oilbeater — APPROVED** [review]

**Pull request merged**